src/main/java/wolox/training/repositories/BookRepository.java

```diff
@@ -11,4 +12,6 @@
     public Optional<Book> findFirstByAuthorIgnoreCase(String author);
 
     public Optional<Book> findByIsbn(String isbn);
+
+    public Optional<List<Book>> findByPublisherAndGenreAndYear(String publisher, String genre, String year);
```

Comment:
When you work with lists you don't need to return an Optional, you can return the List directly because if there are no books, it will be empty, but not null 😄 
Same for the list of users